_plugins/api.rb

```diff
@@ -85,6 +93,38 @@ def self.generate_locations_endpoint(site)
         'Index of team members by location code', data)
     end
 
+    # Based off of
+    # https://github.com/jekyll/jekyll/blob/v2.5.2/test/test_page.rb#L10-L13
+    # Modifies `page`
+    def self.do_render(site, page)
+      layout = Jekyll::Layout.new(site, '_layouts', 'simple.html')
+      layouts = Hash.new(layout)
+      page.render(layouts, site.site_payload)
+    end
+
+    def self.pages(site)
+      site.pages.select {|page| %w(.html .md).include?(page.ext) }
+    end
+
+    def self.get_output(site, page)
+      do_render(site, page)
+      filterer = Filterer.new
+      filterer.condense(filterer.strip_html(page.content))
+    end
```

Comment:
Why do you need to generate the output in the plugin itself? How will it be consumed? I was assuming that the normal Jekyll site generation would take place, then the indexing would happen afterwards.